### PR TITLE
Update segmentation learner flow and visualization 

### LINKED ISSUE #10: Create a base segmentation learner
Need to support 
- Visualization task: 
    - De-normalize transformed rgb images  
    - Color mapping from tensor (as same as PASCAL VOC)

### PR DESCRIPTION
**Pull request by nhtlongcs**
Fix #10

### CODE REVIEW

File: nncore/utils/utils.py [hunk 1]
```diff
@@ -40,7 +40,37 @@ def vprint(obj: str, verbose: bool):
         print(obj)
 
 
-def tensor2plt(obj: Tensor):
+def tensor2plt(obj: Tensor, title: List[Any]):
```
Comment (VinhLoiIT) [inline nncore/utils/utils.py:43]: https://matplotlib.org/3.1.1/api/_as_gen/matplotlib.pyplot.title.html

`title` should be `str`

File: nncore/utils/logger.py [hunk 2]
```diff
@@ -24,10 +24,9 @@ def update_figure(self, tag, image, step):
         self.writer.add_figure(tag, image, step)
 
 
-def image_batch_show(batch, ncol=5, fig_size=(30, 10)):
-    # https://stackoverflow.com/questions/51329159/how-can-i-generate-and-display-a-grid-of-images-in-pytorch-with-plt-imshow-and-t
-    grid_img = torchvision.utils.make_grid(batch, nrow=ncol)
-    return grid_img.float()
+def image_batch_show(batch, ncol=5, fig_size=(30, 10), normalize=False):
```
Comment (VinhLoiIT) [inline nncore/utils/logger.py:27]: what is `fig_size` argument used for?
Moreover, if this does nothing more than calling `torchvision.utils.make_grid`, try define an alias name instead.

File: nncore/models/loss/diceloss.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                        target.shape[1], self.weight.shape[0]
+                    )
+                    dice_loss *= self.weights[i]
+                total_loss += dice_loss
+
+        return total_loss / target.shape[1]
+
+
+class Dicewithstat(nn.Module):
+    r"""Dicewithstat is warper of cross-entropy loss"""
```
Comment (VinhLoiIT) [inline nncore/models/loss/diceloss.py:96]: Please use camel case for class names.
About the documentation string, is it a wrapper of the cross-entropy loss or dice loss?

File: nncore/models/loss/celoss.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+from typing import Dict
+from torch.functional import Tensor
+from torch.nn import functional as F
+
+from torch import nn
+import torch
+
+
+class CEwithstat(nn.Module):
+    r"""CEwithstat is warper of cross-entropy loss"""
```
Comment (VinhLoiIT) [inline nncore/models/loss/celoss.py:10]: Is it typo error here? `warper` -> `wrapper`
Please use camel case for class names

File: nncore/learner/segmentationlearner.py [hunk 5]
```diff
@@ -1,14 +1,16 @@
...
 from ..schedulers import *
+from ..utils.utils import inverse_normalize_batch
 
 from torch.optim import Optimizer
 from torch.utils.data import DataLoader
 from torch.nn import Module
 from torch import device
 
-
+# from ..utils.cityscapes import cityscapes_map
```
Comment (VinhLoiIT) [inline nncore/learner/segmentationlearner.py:13]: Please remove redundant imports if possible.

File: examples/segmentation/train.py [hunk 6]
```diff
@@ -11,5 +11,6 @@
 
 if __name__ == "__main__":
     opt = opts(cfg_path="./config/opt.yaml").parse()
-    train_pipeline = Pipeline(opt)
-    train_pipeline.fit()
+    print(opt)
+    # train_pipeline = Pipeline(opt)
+    # train_pipeline.fit()
```
Comment (VinhLoiIT) [inline examples/segmentation/train.py:16]: why we have this change?

**Review by VinhLoiIT — CHANGES_REQUESTED** [review]
I've added some minor change requests. Please update.
